### ISSUE TITLE
Silent crash on startup if DebugView is running

### ISSUE THREAD

**Issue by riQQ** [smourier/TraceSpy#15]
WpfTraceSpy 4.0.0.5 silently crashes on startup if DebugView 4.90 is running.

```
System.UnauthorizedAccessException: Access to the path 'DBWIN_BUFFER_READY' is denied.
  mscorlib.dll!System.IO.__Error.WinIOError(int errorCode, string maybeFullPath) (IL≈0x012E, Native=0x00007FF8AD5743A0+0x430)
  mscorlib.dll!System.Threading.EventWaitHandle.EventWaitHandle(bool initialState, System.Threading.EventResetMode mode, string name) (IL=0x00AF, Native=0x00007FF8AD573C10+0x2D6)
  WpfTraceSpy.exe!TraceSpy.MainWindow.MainWindow() (IL=0x0185, Native=0x00007FF8AD03E8E0+0x5B4)
  [Native to Managed Transition]
  mscorlib.dll!System.RuntimeType.CreateInstanceSlow(bool publicOnly, bool skipCheckThis, bool fillCache, ref System.Threading.StackCrawlMark stackMark) (IL≈0x006A, Native=0x00007FF8AC97FDE0+0x1D1)
  mscorlib.dll!System.RuntimeType.CreateInstanceDefaultCtor(bool publicOnly, bool skipCheckThis, bool fillCache, ref System.Threading.StackCrawlMark stackMark) (IL≈0x00AB, Native=0x00007FF8AC97FAB0+0x255)
  mscorlib.dll!System.Activator.CreateInstance(System.Type type, bool nonPublic) (IL≈0x003A, Native=0x00007FF8AC97F920+0x160)
  mscorlib.dll!System.RuntimeType.CreateInstanceImpl(System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, object[] args, System.Globalization.CultureInfo culture, object[] activationAttributes, ref System.Threading.StackCrawlMark stackMark) (IL≈0x01BE, Native=0x00007FF8ACF2B9A0+0x6F1)
  mscorlib.dll!System.Activator.CreateInstance(System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, object[] args, System.Globalization.CultureInfo culture, object[] activationAttributes) (IL≈0x00AF, Native=0x00007FF8ACF2B650+0x31C)
  mscorlib.dll!System.Activator.CreateInstance(System.Type type, params object[] args) (IL≈0x0000, Native=0x00007FF8ACF2B5E0+0x47)
  System.Xaml.dll!System.Xaml.Schema.SafeReflectionInvoker.CreateInstanceCritical(System.Type type, object[] arguments) (IL≈0x0000, Native=0x00007FF8AD020950+0x33)
  System.Xaml.dll!System.Xaml.Schema.SafeReflectionInvoker.CreateInstance(System.Type type, object[] arguments) (IL≈0x0007, Native=0x00007FF8AD0205C0+0x53)
  System.Xaml.dll!System.Xaml.Schema.XamlTypeInvoker.CreateInstanceWithActivator(System.Type type, object[] arguments) (IL≈0x0000, Native=0x00007FF8AD03C120+0x37)
  System.Xaml.dll!System.Xaml.Schema.XamlTypeInvoker.CreateInstance(object[] arguments) (IL≈0x0037, Native=0x00007FF8AD03B3F0+0xCD)
  System.Xaml.dll!MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateInstanceWithCtor(System.Xaml.XamlType xamlType, object[] args) (IL≈0x0006, Native=0x00007FF8AD03AD00+0x53)
  System.Xaml.dll!MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateInstance(System.Xaml.XamlType xamlType, object[] args) (IL≈0x0029, Native=0x00007FF8AD03A9C0+0x12D)
  System.Xaml.dll!MS.Internal.Xaml.Runtime.PartialTrustTolerantRuntime.CreateInstance(System.Xaml.XamlType xamlType, object[] args) (IL≈0x0019, Native=0x00007FF8AD03A7B0+0xAB)
  System.Xaml.dll!System.Xaml.XamlObjectWriter.Logic_CreateAndAssignToParentStart(MS.Internal.Xaml.Context.ObjectWriterContext ctx) (IL≈0x0099, Native=0x00007FF8AD039E20+0x2C5)
  System.Xaml.dll!System.Xaml.XamlObjectWriter.WriteStartMember(System.Xaml.XamlMember property) (IL=0x01A1, Native=0x00007FF8ACF0D690+0x643)
  System.Xaml.dll!System.Xaml.XamlWriter.WriteNode(System.Xaml.XamlReader reader) (IL≈0x006C, Native=0x00007FF8ACEFF970+0x19E)
  PresentationFramework.dll!System.Windows.Markup.WpfXamlLoader.TransformNodes(System.Xaml.XamlReader xamlReader, System.Xaml.XamlObjectWriter xamlWriter, bool onlyLoadOneNode, bool skipJournaledProperties, bool shouldPassLineNumberInfo, System.Xaml.IXamlLineInfo xamlLineInfo, System.Xaml.IXamlLineInfoConsumer xamlLineInfoConsumer, MS.Internal.Xaml.Context.XamlContextStack<System.Windows.Markup.WpfXamlFrame> stack, System.Windows.Markup.IStyleConnector styleConnector) (IL=0x023F, Native=0x00007FF8ACEE6530+0x7ED)
  PresentationFramework.dll!System.Windows.Markup.WpfXamlLoader.Load(System.Xaml.XamlReader xamlReader, System.Xaml.IXamlObjectWriterFactory writerFactory, bool skipJournaledProperties, object rootObject, System.Xaml.XamlObjectWriterSettings settings, System.Uri baseUri) (IL=0x00CA, Native=0x00007FF8ACEE2520+0x375)
  PresentationFramework.dll!System.Windows.Markup.WpfXamlLoader.LoadBaml(System.Xaml.XamlReader xamlReader, bool skipJournaledProperties, object rootObject, System.Xaml.Permissions.XamlAccessLevel accessLevel, System.Uri baseUri) (IL≈0x0014, Native=0x00007FF8ACEE2120+0x94)
  PresentationFramework.dll!System.Windows.Markup.XamlReader.LoadBaml(System.IO.Stream stream, System.Windows.Markup.ParserContext parserContext, object parent, bool closeStream) (IL≈0x00FF, Native=0x00007FF8ACED7490+0x445)
  PresentationFramework.dll!System.Windows.Application.LoadBamlStreamWithSyncInfo(System.IO.Stream stream, System.Windows.Markup.ParserContext pc) (IL≈0x0031, Native=0x00007FF8AD038640+0x137)
  PresentationFramework.dll!System.Windows.Application.LoadComponent(System.Uri resourceLocator, bool bSkipJournaledProperties) (IL≈0x004A, Native=0x00007FF8AD037D20+0x137)
  PresentationFramework.dll!System.Windows.Application.DoStartup() (IL≈0x0072, Native=0x00007FF8AD036490+0x1EC)
  PresentationFramework.dll!System.Windows.Application.<.ctor>b__1_0(object unused) (IL=0x0024, Native=0x00007FF8AD035B40+0xAA)
  WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs) (IL≈0x0077, Native=0x00007FF8ACB22DB0+0x1F5)
  WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler) (IL≈0x0002, Native=0x00007FF8ACB22C20+0x5F)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.WrappedInvoke(System.Delegate callback, object args, int numArgs, System.Delegate catchHandler) (IL≈0x0000, Native=0x00007FF8ACB22BA0+0x5E)
  WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeImpl() (IL≈0x007A, Native=0x00007FF8AD0307E0+0x1A3)
  WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(object state) (IL=0x000D, Native=0x00007FF8AD030780+0x44)
  WindowsBase.dll!MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(object obj) (IL=0x0022, Native=0x00007FF8AD030590+0x9F)
  mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) (IL=0x0080, Native=0x00007FF8ACF2ECD0+0x248)
  mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) (IL=0x0009, Native=0x00007FF8ACF2E690+0x47)
  mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state) (IL=0x0034, Native=0x00007FF8ACF2E500+0x110)
  WindowsBase.dll!MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext executionContext, System.Threading.ContextCallback callback, object state) (IL=0x004A, Native=0x00007FF8AD030160+0x124)
  WindowsBase.dll!System.Windows.Threading.DispatcherOperation.Invoke() (IL=0x0020, Native=0x00007FF8AD02FF70+0x69)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.ProcessQueue() (IL=0x00EB, Native=0x00007FF8AD02F090+0x2FC)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.WndProcHook(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled) (IL=0x0045, Native=0x00007FF8AD02EE00+0xF8)
  WindowsBase.dll!MS.Win32.HwndWrapper.WndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled) (IL≈0x0030, Native=0x00007FF8ACB23240+0x117)
  WindowsBase.dll!MS.Win32.HwndSubclass.DispatcherCallbackOperation(object o) (IL≈0x0036, Native=0x00007FF8ACB23100+0x109)
  WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs) (IL≈0x0077, Native=0x00007FF8ACB22DB0+0x1F5)
  WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler) (IL≈0x0002, Native=0x00007FF8ACB22C20+0x5F)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.WrappedInvoke(System.Delegate callback, object args, int numArgs, System.Delegate catchHandler) (IL≈0x0000, Native=0x00007FF8ACB22BA0+0x5E)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority priority, System.TimeSpan timeout, System.Delegate method, object args, int numArgs) (IL≈0x00C1, Native=0x00007FF8AC9B2D90+0x319)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.Invoke(System.Windows.Threading.DispatcherPriority priority, System.Delegate method, object arg) (IL≈0x0010, Native=0x00007FF8AC9B29D0+0x79)
  WindowsBase.dll!MS.Win32.HwndSubclass.SubclassWndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam) (IL≈0x011D, Native=0x00007FF8AC9AB7F0+0x390)
  [Managed to Native Transition]
  WindowsBase.dll!System.Windows.Threading.Dispatcher.TranslateAndDispatchMessage(ref System.Windows.Interop.MSG msg) (IL≈0x0013, Native=0x00007FF8AD02E7F0+0x71)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame frame) (IL=0x0048, Native=0x00007FF8AD02DC60+0xE8)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.PushFrame(System.Windows.Threading.DispatcherFrame frame) (IL=0x0065, Native=0x00007FF8AD02DA90+0x1AF)
  WindowsBase.dll!System.Windows.Threading.Dispatcher.Run() (IL=0x000A, Native=0x00007FF8AD02D930+0x43)
  PresentationFramework.dll!System.Windows.Application.RunDispatcher(object ignore) (IL=0x0024, Native=0x00007FF8AD02D880+0x92)
  PresentationFramework.dll!System.Windows.Application.RunInternal(System.Windows.Window window) (IL≈0x00E6, Native=0x00007FF8AD02D180+0x41D)
  PresentationFramework.dll!System.Windows.Application.Run(System.Windows.Window window) (IL≈0x001D, Native=0x00007FF8AD02D040+0x9A)
  PresentationFramework.dll!System.Windows.Application.Run() (IL≈0x0007, Native=0x00007FF8AD02CFE0+0x36)
  WpfTraceSpy.exe!TraceSpy.App.Main() (IL≈0x000B, Native=0x00007FF8AC5E5840+0x50)
```

Relevant code. Error happens in line 62.
https://github.com/smourier/TraceSpy/blob/d7cf9b75d3911a756ad1c3d85edde7ea6e502dae/WpfTraceSpy/MainWindow.xaml.cs#L60-L74

**Comment by smourier** [comment]
Hi,

Thanks for reporting that, I'm pretty sure it use to pass that sometimes in the past. Anyway, I've made a fix https://github.com/smourier/TraceSpy/commit/79cf45db83b0b6ddbc79a7fd4275cfb450e4c5e3

**Comment by riQQ** [comment]
I can confirm that it's fixed with 4.0.0.6, thanks :+1:

![screenshot](https://github.com/smourier/TraceSpy/assets/3404787/8e785952-12d8-42b6-b872-f36fe90b30e3)
